lib/HLSL/DxilValidation.cpp

```diff
@@ -3237,8 +3239,30 @@ static void ValidateFunctionBody(Function *F, ValidationContext &ValCtx) {
           }
 
           unsigned opcode = OpcodeConst->getLimitedValue();
+          if (opcode >= static_cast<unsigned>(DXIL::OpCode::NumOpCodes)) {
```

Comment:
Is this right or should we verify it doesn't exceed the number of opcodes for the current version? We have some custom checks about such things and if doing that silences them, then we probably shouldn't, but it might be kind of nice to have a broad catchall that saves us the trouble of making those individual checks.

Reply:
We need to verify that the value is in-range before we can cast it to the `DXIL::OpCode` type because casting an out-of-range value is undefined behavior.

Reply:
I'm realizing now that we can't actually use the versioned numopcodes to indicate version support anyway because we consciously decided to allow some intermingling there.